meteor/server/api/rest/v1/showstyles.ts

```diff
@@ -97,17 +110,71 @@ class ShowStylesServerAPI implements ShowStylesRestAPI {
-		const validateOK = validation.messages.reduce((acc, msg) => acc && msg.level === NoteSeverity.INFO, true)
-		if (!validateOK) {
-			const details = JSON.stringify(
-				validation.messages.filter((msg) => msg.level < NoteSeverity.INFO).map((msg) => msg.message.key),
-				null,
-				2
-			)
-			logger.error(`addOrUpdateShowStyleBase failed validation with errors: ${details}`)
-			throw new Meteor.Error(409, `ShowStyleBase ${showStyleBaseId} has failed validation`, details)
-		}
+		checkValidation(`addOrUpdateShowStyleBase ${showStyleBaseId}`, validation.messages)
 
+		// wait for the upsert to complete before upgrade
+		await new Promise<void>((resolve) => setTimeout(() => resolve(), 200))
+		return ClientAPI.responseSuccess(await runUpgradeForShowStyleBase(showStyleBaseId))
+	}
+
+	async getShowStyleConfig(
+		_connection: Meteor.Connection,
+		_event: string,
+		showStyleBaseId: ShowStyleBaseId
+	): Promise<ClientAPI.ClientResponse<object>> {
+		const showStyleBase = await ShowStyleBases.findOneAsync(showStyleBaseId)
+		if (!showStyleBase) throw new Meteor.Error(404, `ShowStyleBase ${showStyleBaseId} does not exist`)
+
+		return ClientAPI.responseSuccess((await APIShowStyleBaseFrom(showStyleBase)).config)
+	}
+
+	async updateShowStyleConfig(
+		_connection: Meteor.Connection,
+		_event: string,
+		showStyleBaseId: ShowStyleBaseId,
+		config: object
+	): Promise<ClientAPI.ClientResponse<void>> {
+		const existingShowStyleBase = await ShowStyleBases.findOneAsync(showStyleBaseId)
+		if (existingShowStyleBase) {
+			const rundowns = (await Rundowns.findFetchAsync(
+				{ showStyleBaseId },
+				{ projection: { playlistId: 1 } }
+			)) as Array<Pick<Rundown, 'playlistId'>>
+			const playlists = (await RundownPlaylists.findFetchAsync(
+				{ _id: { $in: rundowns.map((r) => r.playlistId) } },
+				{
+					projection: {
+						activationId: 1,
+					},
+				}
+			)) as Array<Pick<DBRundownPlaylist, 'activationId'>>
+			if (playlists.some((playlist) => playlist.activationId !== undefined)) {
+				throw new Meteor.Error(
+					412,
+					`Cannot update ShowStyleBase ${showStyleBaseId} as it is in use by an active Playlist`
+				)
+			}
+		} else throw new Meteor.Error(404, `ShowStyleBase ${showStyleBaseId} not found`)
+
+		const apiShowStyleBase = await APIShowStyleBaseFrom(existingShowStyleBase)
+		apiShowStyleBase.config = config
+
+		const blueprintConfigValidation = await validateAPIBlueprintConfigForShowStyle(
+			apiShowStyleBase,
+			protectString(apiShowStyleBase.blueprintId)
+		)
+		checkValidation(`updateShowStyleConfig ${showStyleBaseId}`, blueprintConfigValidation)
+
+		const showStyle = await showStyleBaseFrom(apiShowStyleBase, showStyleBaseId)
+		if (!showStyle) throw new Meteor.Error(400, `Invalid ShowStyleBase`)
+
+		await ShowStyleBases.upsertAsync(showStyleBaseId, showStyle)
+
+		const validation = await validateConfigForShowStyleBase(showStyleBaseId)
+		checkValidation(`updateShowStyleConfig ${showStyleBaseId}`, validation.messages)
```

Comment:
Why do we do this validation _after_ the `upsertAsync` to write to db? Wouldn't it have been better to have made this validation before the write?
Or am I missing the point? (If so, perhaps add a comment to the code explaining why?)